### PR TITLE
HDDS-7443. CLI admin namespace -quota should return whole file system disk capacity consumed by all replicas

### PR DESCRIPTION
**Pull request by DaveTeng0**
## What changes were proposed in this pull request?
HDDS-7443. CLI admin namespace -quota should return whole file system disk capacity consumed by all replicas

Command `ozone admin namespace quota ` should return whole file system's disk capacity for root/volume/bucket.
(HDFS -df command returns whole file system's disk capacity for directory)

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7443

## How was this patch tested?
- Manual test.

### CODE REVIEW

**Comment by DaveTeng0** [comment]
cc. @jojochuang  @umamaheswararao @swagle @arp7  @kerneltime

File: hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithFSO.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import org.apache.hadoop.hdds.scm.container.ContainerNotFoundException;
 import org.apache.hadoop.hdds.scm.container.ContainerReplica;
 import org.apache.hadoop.hdds.scm.server.OzoneStorageContainerManager;
+import org.apache.hadoop.hdds.scm.container.placement.metrics.SCMNodeStat;
```
Comment (adoroszlai) [inline hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithFSO.java:30]: It seems this import is missing from `TestNSSummaryEndpointWithLegacy`, causing build failure.

https://github.com/apache/ozone/actions/runs/3481884746/jobs/5823544988#step:6:2109

Comment (DaveTeng0) [reply hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithFSO.java:30]: Thanks Atila for the quick review! updated!

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/RootEntityHandler.java [hunk 2]
```diff
@@ -129,23 +130,10 @@ public QuotaUsageResponse getQuotaResponse()
...
-    long quotaInBytes = 0L;
-    long quotaUsedInBytes = 0L;
 
-    for (OmVolumeArgs volume: volumes) {
-      final long quota = volume.getQuotaInBytes();
-      assert (quota >= -1L);
-      if (quota == -1L) {
-        // If one volume has unlimited quota, the "root" quota is unlimited.
-        quotaInBytes = -1L;
-        break;
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/RootEntityHandler.java:141]: The original way to calculate the disk capacity under 'root' will return -1, if any of the volume's quota is not set (means, 'unlimited' quota, default value is -1), which can't show the user what's the current available disk capacity from whole file system's perspective.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/VolumeEntityHandler.java [hunk 3]
```diff
@@ -117,21 +116,17 @@ public DUResponse getDuResponse(
...
-    long quotaInBytes = volumeArgs.getQuotaInBytes();
-    long quotaUsedInBytes = 0L;
+    RootEntityHandler rootEntityHandler = new RootEntityHandler(
+        getReconNamespaceSummaryManager(),
+        getOmMetadataManager(),
+        getReconSCM(),
+        "/"
+    );
+    QuotaUsageResponse rootQuotaUsageResponse = 
+        rootEntityHandler.getQuotaResponse();
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/VolumeEntityHandler.java:126]: in HDFS, -df returns capacity of the whole file system when df towards a directory, which we'll do similar here in Ozone.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketEntityHandler.java [hunk 4]
```diff
@@ -126,15 +125,17 @@ public DUResponse getDuResponse(
...
-    long quotaUsedInBytes = getTotalSize(bucketObjectId);
-    quotaUsageResponse.setQuota(quotaInBytes);
-    quotaUsageResponse.setQuotaUsed(quotaUsedInBytes);
+    RootEntityHandler rootEntityHandler = new RootEntityHandler(
+        getReconNamespaceSummaryManager(),
+        getOmMetadataManager(),
+        getReconSCM(),
+        "/"
+    );
+    QuotaUsageResponse rootQuotaUsageResponse =
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketEntityHandler.java:134]: in HDFS, -df returns capacity of the whole file system when df towards a directory, which we'll do similar here in Ozone for bucket.

**Pull request closed without merging**